snuba/utils/streams/backends/kafka.py

```diff
@@ -617,24 +619,55 @@ def closed(self) -> bool:
 DEFAULT_QUEUED_MIN_MESSAGES = 10000
 
 
+@dataclass
+class KafkaBrokerConfig:
+    config: Mapping[str, Any]
```

Comment:
I originally meant a dataclass with each field we allow to be explicitly declared (so we would have a list of allowed config parameters), but I did not realize that we do absolutely nothing with that except passing it to the kafka api, so you are right, that was a bad idea. Sorry about that.

So I would go for the original idea of using a dictionary and we can turn into a typed dict later. Also we do not need to have a list of allowed fields.